src/rendergraph/common/rendergraph/geometry.h

```diff
@@ -0,0 +1,85 @@
+#pragma once
+
+#include <QVector2D>
+#include <QVector3D>
+
+#include "backend/basegeometry.h"
+#include "rendergraph/attributeset.h"
+
+namespace rendergraph {
+class Geometry;
+} // namespace rendergraph
+
+class rendergraph::Geometry : public rendergraph::BaseGeometry {
+  public:
+    struct Point2D {
+        QVector2D position2D;
+        Point2D(float x, float y)
+                : position2D{x, y} {
+        }
```

Comment:
is there an advantage of the constructor instead of just leaving the `struct` as an [aggregate](https://en.cppreference.com/w/cpp/language/aggregate_initialization)?

Reply:
No, I’ll change it.